Mathlib/NumberTheory/DirichletCharacter/Basic.lean

```diff
@@ -114,4 +116,73 @@ lemma mem_conductorSet_dvd {x : ℕ} (hx : x ∈ conductorSet χ) : x ∣ n := h
 The Dirichlet character `χ` can then alternatively be reformulated on `ℤ/nℤ`. -/
 noncomputable def conductor : ℕ := sInf (conductorSet χ)
 
+lemma conductor_mem_conductorSet : conductor χ ∈ conductorSet χ :=
+  Nat.sInf_mem (Set.nonempty_of_mem (level_mem_conductorSet χ))
+
+lemma conductor_dvd_level : conductor χ ∣ n := (conductor_mem_conductorSet χ).dvd
+
+lemma FactorsThrough_conductor : FactorsThrough χ (conductor χ) := conductor_mem_conductorSet χ
+
```

Comment:
I suggest to add the following API lemmas at suitable places.
```lean
lemma level_one (χ : DirichletCharacter R 1) : χ = 1 := by
  ext
  simp [units_eq_one]

lemma level_one' (hn : n = 1) : χ = 1 := by
  subst hn
  exact level_one _

instance : Subsingleton (DirichletCharacter R 1) := by
  refine Iff.mpr subsingleton_iff (fun χ χ' ↦ ?_)
  simp [level_one]

noncomputable instance : Inhabited (DirichletCharacter R n) := ⟨1⟩

noncomputable instance : Unique (DirichletCharacter R 1) := Unique.mk' (DirichletCharacter R 1)

lemma changeLevel_trivial {d : ℕ} (h : d ∣ n) :
    changeLevel h (1 : DirichletCharacter R d) = 1 := by
  simp [changeLevel]

lemma FactorsThrough_one_iff : FactorsThrough χ 1 ↔ χ = 1 := by
  refine ⟨fun ⟨_, χ₀, hχ₀⟩ ↦ ?_,
          fun h ↦ ⟨one_dvd n, 1, by rw [h, changeLevel_trivial]⟩⟩
  rwa [level_one χ₀, changeLevel_trivial] at hχ₀
```

Reply:
Maybe we should discuss whether the trivial character should be referred to as `trivial` or as `one`.
The latter could potentially lead to confusion when level 1 is involved.

Reply:
I added this.

Not sure if `trivial` or `one` or `1` would be best. Let's stick with `1` for now?

Reply:
What I meant was in names of lemmas, i.e., `changeLevel_trivial` vs. `changeLevel_one`. Writing `1` in the *statements* is perfectly fine.

I guess, right now, it should perhaps be `changeLevel_one` for consistency (or else `factorsThrough_trivial`, `conductor_trivial`, `eq_trivial_iff_conductor_eq_one` etc.). I'm to blame for using `trivial` here, so feel free to change it.

Reply:
changed to `changeLevel_one`